### PR TITLE
refresh pluto for 23v2.1

### LINKED ISSUE #165: PLUTO build version: 23v2.1
## Update version name and source data versions

PLUTO uses version numbering YYvMAJOR.MINOR
1. YY for the last two digits of the release year
2. MAJOR version for using the latest versions of all input data
2. MINOR version for using the latest versions of particular input data

Make sure version name is updated in this file
- [x] <https://github.com/NYCPlanning/db-pluto/blob/main/pluto_build/version.env>

In that file, the following are also specified. These versions are to be help constant with what was used in the last major pluto update.
They are not used by the main/major data loading script, so they are only updated at the first minor release of each major release.
For example, say 22v1.4 is the last release. 22v2 is to be built. These args are not used in the build process, so they do not need to be updated.
When it's time to build 22v2.1, these must be updated to match [the versions used in 22v2](https://nyc3.digitaloceanspaces.com/edm-publishing/db-pluto/22v2/latest/output/source_data_versions.csv) as they were last used for 22v1 minor versions. Finally, when 22v2.2 is being built, they don't need to be updated since they've
already been updated to be set to 22v2 versions

- [x] DOF_WEEKLY_DATA_VERSION
- [x] DOF_CAMA_DATA_VERSION

- [x] GEOSUPPORT_VERSION
- [x] FEMA_FIRPS_VERSION
- [x] DOITT_DATA_VERSION
- [x] DOF_DATA_VERSION

- [x] DCP_COLP_VERSION
- [x] DPR_GREENTHUMB_VERSION
- [x] DSNY_FREQUENCIES_VERSION
- [x] LPC_HISTORIC_DISTRICTS_VERSION
- [x] LPC_LANDMARKS_VESRSION

- [x] PLUTO_CORRECTIONS_VERSION

### Updated with Zoning Taxlots 

(check [here](https://github.com/NYCPlanning/db-zoningtaxlots/actions/workflows/dataloading.yml) for latest run).

These are all produced by GIS, who typically update them sometime in the first week of each month.
Check in with them before archiving with data library

- [x] **dof_dtm**
- [x] **dof_shoreline**
- [x] **dof_condo**
- [x] **dcp_commercialoverlay**
- [x] **dcp_limitedheight**
- [x] **dcp_zoningdistricts**
- [x] **dcp_specialpurpose**
- [x] **dcp_specialpurposesubdistricts**
- [x] **dcp_zoningmapamendments**
- [x] **dcp_edesignation**

### Update QAQC App 

~~- [ ] **update data-engineering-qaqc version_comparison_report in src/pluto/pluto.py with version of pluto build** (must be updated with every version of pluto [here] (https://github.com/NYCPlanning/data-engineering-qaqc/blob/main/src/pluto/pluto.py))~~

## Build CI Runs

- **[INSERPT LINKS TO RELEVANT CI RUNS HERE]**

### PR DESCRIPTION
**Pull request by fvankrieken**
Closes #165 - ticket for build itself.

Also addresses #92 twofold
- First, removes the logic around the corrections logic. This failed because of corrections machinery not being able to handle the corrected (original) value being null I believe
- New logic in zoning_zoningdistricts.sql and zoning_specialdistricts.sql. I also merged zoning_zoningdistricts_create.sql and zoning_zoningdistricts.sql - going commit by commit might be clearer on what the changes are. I'll also try to summarize changes in logic and add helpful comments to the sql itself describing why things are being done

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pluto_update_minor.md
Comment (fvankrieken) [inline .github/ISSUE_TEMPLATE/pluto_update_minor.md:32]: Just renaming things to align with their names in edm-recipes. Same thing done in version.env

**Comment by fvankrieken** [comment]
Some discussion with @AmandaDoyle about whether my implementation of fix for #92 should be in this minor version rather than waiting for the next major release. Locally, I've done some comparisons which are summarized here. Changes in results fall into a couple categories, with a couple reasons. Reasons include:
- grouping zoning districts prior to assigning ranking. If a lot has 40% in an R8, 30% in two different PARK districts, before it would be assigned to R8 then PARK, now it would aggregate to 40% and 60% and therefor be assigned PARK as zonedist1

### zonedist1, zonedist2 flipped
 |   bbl     | zd1_old | zd1_new | zd2_old | zd2_new | 
|------------|---------|---------|---------|---------|
| 4151000400 | R3-2    | PARK    | PARK    | R3-2    |  
| 1021400011 | R8      | PARK    | PARK    | R8      | 

[4151000400](https://zola.planning.nyc.gov/l/lot/4/15100/400?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#14.89/40.63422/-73.82389):

image
The two park sections total more area than the R3-2 of the street. Flip caused by grouping zoning districts

[1021400011](https://zola.planning.nyc.gov/l/lot/1/2140/11?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#14.87/40.84333/-73.93863):

image

Many park segments were split by R8 road segment. Flip caused by grouping zoning districts

### zonedist2, zonedist3 flipped
|    bbl     | zd2_old | zd2_new | zd3_old | zd3_new |
|------------|-|-|-|----------|
| 3045860001 | R6      | R3-2    | R3-2    | R6 |

image
The two R3-2 sections total more area than the R6 (largest R3-2 and the r6 are within 2% in area of each other). Flip caused by grouping zoning districts

### new zonedist2 where none assigned before
|  bbl     | zd1_old | zd1_new | zd2_old | zd2_new |
|-|-|-|-|-|
| 1016520001 | R7-2    | R7-2    |         | PARK |
| 2059111140 | R2      | R2      |         | R4 |
| 3020230001 | M3-1    | M3-1    |         | M2-1 |
| 4016000001 | R5      | R5      |         | R6B |
| 4158620001 | C4-4    | C4-4    |         | M1-4/R6 |

[1016520001](https://zola.planning.nyc.gov/l/lot/1/1652/1?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#17.34/40.788944/-73.943829):

image
Two small park zoning districts grouped where neither met threshold before

[2059111140](https://zola.planning.nyc.gov/l/lot/2/5911/1140?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#19.27/40.8933601/-73.9086202):
image
A little more surprising, but two small R4 segments which grouped meet the threshold

[3020230001](https://zola.planning.nyc.gov/l/lot/3/2023/1?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#14.61/40.70054/-73.96645):

image
Multiple sizable M2-1 districts now meet the threshold

[4016000001](https://zola.planning.nyc.gov/l/lot/4/1600/1?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#16.36/40.743748/-73.868985):

image
Small segments at end are each in R6-B

[4158620001](https://zola.planning.nyc.gov/l/lot/4/15862/1?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22tax-lots%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#16.13/40.592539/-73.767452):

image
The whole reason we're here! Special M1-4/R6 district is below 10% threshold but not associated with any other lot. 

### New zonedist3 when null before
|    bbl     | zd3_old | zd3_new |
|-|-|-|
| 2033990001 |         | PARK |
| 5026650020 |         | R3-2 |

[2033990001](https://zola.planning.nyc.gov/l/lot/2/3399/1?layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22zoning-districts%22%5D&search=true):

image
Multiple small park segments that combine to reach threshold

[5026650020](https://zola.planning.nyc.gov/l/lot/5/2665/20?aerial-year=aerials-2016&layer-groups=%5B%22building-footprints%22%2C%22commercial-overlays%22%2C%22street-centerlines%22%2C%22subway%22%2C%22zoning-districts%22%5D&print=false&search=true&selectedCouncilDistricts=%5B%222013%22%5D&selectedFirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedOverlays=%5B%22C1-1%22%2C%22C1-2%22%2C%22C1-3%22%2C%22C1-4%22%2C%22C1-5%22%2C%22C2-1%22%2C%22C2-2%22%2C%22C2-3%22%2C%22C2-4%22%2C%22C2-5%22%5D&selectedPfirm=%5B%22A%22%2C%22Shaded%20X%22%2C%22V%22%5D&selectedZoning=%5B%22BP%22%2C%22C1%22%2C%22C2%22%2C%22C3%22%2C%22C4%22%2C%22C5%22%2C%22C6%22%2C%22C7%22%2C%22C8%22%2C%22M1%22%2C%22M2%22%2C%22M3%22%2C%22PA%22%2C%22R1%22%2C%22R10%22%2C%22R2%22%2C%22R3%22%2C%22R4%22%2C%22R5%22%2C%22R6%22%2C%22R7%22%2C%22R8%22%2C%22R9%22%5D&shouldRefresh=false#14.39/40.59009/-74.17463):

image
Multiple smaller r3-2 segments (road being one of them) combine to reach threshold

### New zonedist4 when null before
 |   bbl      | zd3_old | zd3_new | zd4_old | zd4_new |
|-|-|-|-|-|
 |5026100027 | R3A     | R3-2    |         | R3A|
image
Right next to the last one. The road and smaller zone on the SW side of lot are both R3-2, they combine to both reach threshold and also rank higher than the R3A segment on west corner of lot

### other zonedist issues?
No other cases where 
- zonedist not null and changed
- zonedist was null before and now isn't
- zonedist was assigned before and is now null

Which should be pretty exhaustive

**Comment by fvankrieken** [comment]
Special districts are simpler, both in terms of the code changes as well as the effects. There's no grouping to be done since special districts are unique.

We had two cases where spdist1 was null before because relevant spdist was assigned to spdist2 instead. Those are now fixed (bbl 2000008900 and 5004870100). Then we have cases where special districts are being assigned (as spdist1) to lots where previously no spdist was assigned... 

And it's just our favorite lot, from #92! MX-21 is successfully associated where it wasn't before

image

No other changes introduced

**Comment by fvankrieken** [comment]
Awaiting final word on ordering of G/MX-11 and EHC/TA special districts, but other than that could use review at this point. Happy to add code comments where sql is unclear

File: products/pluto/pluto_build/sql/corr_zoning.sql
Comment (alexrichey) [inline products/pluto/pluto_build/sql/corr_zoning.sql:1]: Just curious, why weren't these applied?

Comment (fvankrieken) [reply products/pluto/pluto_build/sql/corr_zoning.sql:1]: Corrections currently don't allow replacing "null" values in terms of how the sql is written, just replacing present but incorrect values. That would be super easy to change, but per discussions with Amanda and Matt, we'd prefer to not fix district "issues" with corrections anyways as they'd sort of have to be maintained (whenever lots, zoning boundaries updated, is the correction still valid etc)

**Comment by alexrichey** [comment]
@fvankrieken Just FYI, still reading through. Just want to make sure I fully understand, but makes sense so far. The documentation on changes is really excellent though!

File: products/pluto/pluto_build/sql/zoning_specialdistrict.sql
Comment (alexrichey) [inline products/pluto/pluto_build/sql/zoning_specialdistrict.sql:4]: @fvankrieken is this comment accurate? (or was it ever?) Isn't is the other way around? ie 10% of the lot must be covered by the district. Just checking my understanding.

Comment (fvankrieken) [reply products/pluto/pluto_build/sql/zoning_specialdistrict.sql:4]: You are correct! Missworded comment

File: products/pluto/pluto_build/version.env
Comment (damonmcc) [inline products/pluto/pluto_build/version.env:1]: curious about the source data version changes here (like `DPR_GREENTHUMB_VERSION`)

if it's for a minor build, don't we want to leave those the same as 23v2?

Comment (fvankrieken) [reply products/pluto/pluto_build/version.env:1]: These variables aren't updated as part of a major build, so before any changes they correspond to 23v1

Comment (damonmcc) [reply products/pluto/pluto_build/version.env:1]: ahhh I see, so this file becomes kinda misleading right after a major build since the versions are for the major before the latest

File: products/pluto/pluto_build/sql/zoning_zoningdistrict.sql
Comment (alexrichey) [inline products/pluto/pluto_build/sql/zoning_zoningdistrict.sql:40]: @fvankrieken (cc @damonmcc ) potentially silly question, but is ST_Intersection not commutative? Wondering why we have both `segbblgeom` and `segzonegeom`

Comment (fvankrieken) [reply products/pluto/pluto_build/sql/zoning_zoningdistrict.sql:40]: That is a great point, it definitely is

**Review by damonmcc — APPROVED** [review]
lgtm!

it's a nit but since you'll probably rebase rather than squash, is that commit message `f special district - fix comment` a typo at the beginning?

**Review by alexrichey — APPROVED** [review]
Looks good!

**Pull request merged**
